src/Parallel/Algorithms/AlgorithmArray.hpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// This file should be included in each file which defines an array
+/// parallel component; doing so ensures that the correct charm++ chares are
+/// defined for executables that use that parallel component.
+
+#pragma once
+
+#include "Parallel/Algorithm.hpp"
+#include "Parallel/Algorithms/AlgorithmArrayDeclarations.hpp"
+#include "Parallel/ArrayIndex.hpp"
+
+/*!
+ * \ingroup ParallelGroup
+ * \brief A charm++ algorithm array chare.
+ *
+ * \details This object is the direct definition of the charm++ object
```

Comment:
Help, I don't know what it means for an object to be the "direct definition" of another object...

(Same comment for the other algorithm types)

Reply:
Agreed, this is unnecessarily fuzzy wording. What I mean here is "This is the definition of the implementation object for a charm++ chare, indicated by the object name that is defined in the accompanying .ci file". Is that clearer? Is there a better rephrasing that you can think of?

Reply:
How about `is the definition of the distributed Charm++ object...`. It would also be good to say explicitly (somewhere in this comment, maybe in the next paragraph) that "`AlgorithmArray` is strong typedef so that the core functionality..." (looks like the below paragraph is the right place). I'm basically looking for the term "strong typedef"